wgsl/index.bs

```diff
@@ -2544,51 +2547,103 @@ programmable parts of pipelines:
 * <dfn noexport>vertex</dfn>
 * <dfn noexport>fragment</dfn>
 
-<pre class='def'>
-pipeline_stage
-  : COMPUTE
-  | VERTEX
-  | FRAGMENT
-</pre>
-
 Each shader stage has its own set of features and constraints, described elsewhere.
 
-## Entry point declaration TODO ## {#entry-point-decl}
+## Entry point declaration ## {#entry-point-decl}
 
-The `entry_point` declares an entry point into the module. The entry points may
-be forward declarations but the functions referenced must be declared in the
-file.
+An <dfn noexport>entry point</dfn> is a function that is invoked to perform
+the work for a particular [=shader stage=].
 
-The input and output parameters to the entry point are determined by which
-global variables are used in the function and any called functions.
+Specify a `stage` attribute on a function declaration to declare that function
+as an entry point.
 
-The entry point function must be declared with no arguments and returning `void`.
+When configuring the stage in the pipeline, the entry point is specified by providing
+the [SHORTNAME] module and the entry point's function name.
 
-<pre class='def'>
-entry_point_decl
-  : ENTRY_POINT pipeline_stage EQUAL IDENT
-  | ENTRY_POINT pipeline_stage AS STRING_LITERAL EQUAL IDENT
-  | ENTRY_POINT pipeline_stage AS IDENT EQUAL IDENT
-</pre>
+An entry point function must have no parameters,
+and its return type must be [=void=].
+
+ISSUE: We've lost the ability to use shader names that are invalid identifiers.
```

Comment:
Do we want that? I think all shader entry point names should be valid identifiers.

Reply:
It would be best for bijectivity with SPIR-V.

Also, if I'm reading this correctly it removes the entry point name override entirely. Wasn't it the case that it was valid to have `stage=vertex,name=main` and `stage=fragment,name=main`? Now it's not possible because the two functions would have the same name.

Reply:
Unless entry point functions (`[[stage vertex]] fn`) are not callable from other functions.

Reply:
You could previously have multiple functions with the same name through the entry point name. I'm not sure if we need to support that. SPIR-V can always canonicalize to change the entry point name to match the function.

Entry point functions aren't callable by other functions.

Reply:
I think MSL and HLSL require that entry points be identifiers anyway.  So we have to restrict the SPIR-V functionality anyway.

Reply:
Right, it removes the name override entirely.  That's another restriction.

Reply:
I've deleted this issue in the proposed text.